### PR TITLE
Add secondaries documentation

### PR DESCRIPTION
**Pull request by Jean1995**
While the documentation of our configuration file is quite detailed, there has been only very little documentation of the `Secondaries` class, which is the output of the `Propagator` class. 
However, this class is very important for almost every user.

Therefore, I wrote some documentation for all relevant methods to both C++ and Python.

Furthermore, I added two new functions: `final_state()` and `initial_state()` to get the initial and final particle state of a particle out of the Secondaries class. They are probably more intuitive to use than `track()[-1]` and `track()[0]` which had to be used before.
The jupyter notebook have been adapted accordingly to use these new functions.

### CODE REVIEW

**Comment by Jean1995** [comment]
CI is somehow busted at the moment...

**Comment by sudojan** [comment]
> ... get the initial and final particle state of a particle out of the Secondaries class.

You mean "out of the Track"

File: examples/AdvancedPropagator.ipynb [hunk 1]
```diff
@@ -188,7 +188,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "100%|██████████| 100000/100000 [00:17<00:00, 5654.12it/s]\n"
+      "100%|██████████| 100000/100000 [00:01<00:00, 52909.55it/s]\n"
```
Comment (sudojan) [inline examples/AdvancedPropagator.ipynb:191]: is there a reason why it is so much faster?

**Comment by Jean1995** [comment]
> > ... get the initial and final particle state of a particle out of the Secondaries class.
> 
> You mean "out of the Track"

Well, basically both: The Secondaries class contains/manages the track object. However, the idea is that users shouldn't need to understand the track object to get simple information out of the propagator (especially energy losses or the final particle state).

**Review by sudojan — APPROVED** [review]

**Pull request merged**